info-compiler/src/main/java/com/google/sps/webcrawler/WebCrawler.java

```diff
@@ -176,15 +176,24 @@ public void compileNewsArticle(String candidateName, String candidateId) {
   List<NewsArticle> extractUrlsAndMetadataFromCustomSearchJson(JsonObject json) {
     List<NewsArticle> newsArticles = new ArrayList<>(CUSTOM_SEARCH_RESULT_COUNT);
     JsonArray searchResults = json.getAsJsonArray("items");
+    if (searchResults == null) {
+      return Arrays.asList();
+    }
     int priority = 1;
     for (JsonElement result : searchResults) {
-      JsonObject metadata =
-          (JsonObject)
-              (((JsonObject) result)
-                   .getAsJsonObject("pagemap")
-                       .getAsJsonArray("metatags")
-                           .get(0));
-      String url = metadata.get(CUSTOM_SEARCH_URL_METATAG).getAsString();
+      JsonObject metadata;
+      String url;
+      try {
+        metadata =
+            (JsonObject)
```

Comment:
what would this return if you didn't cast twice?

Reply:
There has to be two casts.
JsonObject inherits JsonElement and JsonObject has additional methods.
`result` is a JsonElement, meaning that it doesn't have the method `.getAsJsonObject()`. A cast is required. `.get(0)` also returns a JsonElement, but since we want `metadata` to be a JsonObject, the second cast is required.